types/src/transaction/authenticator.rs

```diff
@@ -1569,4 +1664,527 @@ mod tests {
         let verification_result = signed_txn.verify_signature();
         assert!(verification_result.is_ok());
     }
+
+    #[test]
```

Comment:
These tests could probably be made much shorter if you have the right helper functions.

Also, I don't think hardcoding the base64 JWT is a good idea: it will make updating the tests very hard.

Can you instead use a JSON library to create the base64-encoding of the JWT? There should be a function that returns such an encoding. Maybe by default it returns a specific JWT, but there are parameters to change its fields.